src/server/api/models/docgen/service.js

```diff
@@ -0,0 +1,184 @@
+            acc.push(resolved)
+          }
+        }
+        return acc
+      }, [])
+      .map(definition => resolveDefinition(definition, types))
+
+    if (definitions.length === 0) {
+      return false
+    }
+    definitions.forEach((definition) => {
+      if (definition && components.indexOf(definition) === -1) {
+        components.push(definition)
+      }
+    })
+    return false
+  }
+
+  const assignmentExpression = (path) => {
+    let resolvedPath = path
+    // Ignore anything that is not `exports.X = ...;` or
+    // `module.exports = ...;`
+    if (!utils.isExportsOrModuleAssignment(resolvedPath)) {
+      return false
+    }
+    // Resolve the value of the right hand side. It should resolve to a call
+    // expression, something like React.createClass
+    resolvedPath = utils.resolveToValue(resolvedPath.get('right'))
+    if (!isComponentDefinition(resolvedPath)) {
+      resolvedPath = utils.resolveToValue(resolveHOC(resolvedPath))
+      if (!isComponentDefinition(resolvedPath)) {
+        return false
+      }
+    }
+    const definition = resolveDefinition(resolvedPath, types)
+    if (definition && components.indexOf(definition) === -1) {
+      components.push(definition)
+    }
+    return false
+  }
+
+  recast.visit(ast, {
+    visitFunctionDeclaration: false,
+    visitFunctionExpression: false,
+    visitClassDeclaration: false,
+    visitClassExpression: false,
+    visitIfStatement: false,
+    visitWithStatement: false,
+    visitSwitchStatement: false,
+    visitCatchCause: false,
+    visitWhileStatement: false,
+    visitDoWhileStatement: false,
+    visitForStatement: false,
+    visitForInStatement: false,
+    visitExportDeclaration: exportDeclaration,
+    visitExportNamedDeclaration: exportDeclaration,
+    visitExportDefaultDeclaration: exportDeclaration,
+    visitAssignmentExpression: assignmentExpression,
+  })
+
+  return components
+}
+
+// FIXME not all cases of exported elements managed
+// just : export { default } from './xxx'
+const indexResolver = (ast, recast) => {
+  const exportDefaultIndex = (path) => {
+    const importType = {
+      path: path.value.source.value,
+    }
+    throw importType
+  }
+  recast.visit(ast, {
+    visitFunctionDeclaration: false,
+    visitFunctionExpression: false,
+    visitClassDeclaration: false,
+    visitClassExpression: false,
+    visitIfStatement: false,
+    visitWithStatement: false,
+    visitSwitchStatement: false,
+    visitCatchCause: false,
+    visitWhileStatement: false,
+    visitDoWhileStatement: false,
+    visitForStatement: false,
+    visitForInStatement: false,
+    visitExportDeclaration: false,
+    visitExportNamedDeclaration: exportDefaultIndex,
+    visitExportDefaultDeclaration: exportDefaultIndex,
+    visitAssignmentExpression: false,
+  })
+}
+
+const resolvePath = async (path) => {
+  const extensions = ['.js', '.jsx', '']
+  for (let i = 0; i < extensions.length; i += 1) {
```

Comment:
forbidden by the linter : 
`[eslint] iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations. (no-restricted-syntax)`

here the airbnb explanation : 
https://github.com/airbnb/javascript/issues/1271